### PR TITLE
Add Kafka

### LINKED ISSUE #37: Add Kafka to ansible-hosts template

### PR DESCRIPTION
**Pull request by Nuttymoon**
Fixes #37 

Depends on:
- ~~https://github.com/TOSIT-IO/tdp-collection/pull/341~~
- https://github.com/TOSIT-IO/tdp-collection/pull/349
- #41

### CODE REVIEW

**Review by gboutry — APPROVED** [review]

**Review by gboutry — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -8,6 +8,11 @@ Please check the main [Ansible TDP Roles](https://github.com/TOSIT-FR/ansible-td
 
 - `airflow`: deploys the Apache Airflow release (Airflow Webserver + Scheduler + Worker)
 - `livy.server`: deploys Apache Livy Server
+- `kafka`: deploys Kafka brokers and Kafka client
+
+## Other services
+
+- `zookeeper-kafka`: deploys Apache ZooKeeper dedicated to Kafka (version 3.5.9 by default). Note that this service is deployed to hosts member of the `[zk]` group and uses `tdp-collection`'s [zookeeper role](https://github.com/TOSIT-IO/tdp-collection/tree/master/roles/zookeeper).
```
[the remark this answers is not in the document]
Comment (Nuttymoon) [reply README.md:15]: As the `zookeeper-kafka` service uses the `zookeeper` role of `tdp-collection` I don't know if we can really change the behavior. The long-term goal for the Kafka integration to TDP would be to use the same ZooKeeper cluster as the other TDP services.  
On the performance part, ZooKeeper Server is a Java process and therefore cannot overflow configured limits. I would say it is up to the user to control its ZooKeeper instances with the Java `-Xmx` option.

[the remark this answers is not in the document]
Comment (gboutry) [reply README.md:15]: The group should only be mentionned in the playbooks, not the roles, therefore it should be safe to create a different `zk` and `zk_client` group.

[the remark this answers is not in the document]
Comment (rpignolet) [reply README.md:15]: Ok we can keep like that.

**Review by rpignolet — APPROVED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nuttymoon) [reply README.md:15]: @gboutry the problem is that some references to the `zk` group are hard coded:
- [server/tasks/install.yml#L23](https://github.com/TOSIT-IO/tdp-collection/blob/master/roles/zookeeper/server/tasks/install.yml#L23)
  ```yml
  - name: Get an ID for host
    set_fact:
      zk_id: "{{ id + 1  }}"
    loop: "{{ groups['zk'] }}"
    when: inventory_hostname == item
    loop_control:
      index_var: id
  ```
- [check/tasks/main.yml#41](https://github.com/TOSIT-IO/tdp-collection/blob/fa68888b13f1aa68c8cc096049e0b333bd27d714/roles/zookeeper/check/tasks/main.yml#L41)
  ```yml
  - name: ZK service check - Fetch server mode
    ansible.builtin.shell: /usr/bin/echo srvr | /usr/bin/nc {{ item }} {{ zookeeper_server_client_port }} | /usr/bin/awk '$1=="Mode:" {print $2}'
    register: zk_server_mode
    loop: "{{ groups['zk'] }}"
  ```

**Pull request merged**